akka-http/src/main/scala/akka/http/scaladsl/server/PathMatcher.scala

```diff
@@ -358,6 +358,22 @@ trait PathMatchers {
 
   /**
    * A PathMatcher that matches and extracts the complete remaining,
+   * unmatched part of the request's URI path as an decoded (with actual whitespace) String.
+   * @group pathmatcher
+   */
+  object RemainingDecoded extends PathMatcher1[String] {
+    private val encodedSpace = "%20"
+    private val decodedSpace = " "
+
+    def decodeWhitespace(path: String) = path.replace(encodedSpace, decodedSpace)
```

Comment:
Moreover off top question: why implicit operations for tuples (`TupleOps`) are set up for any type `T`? Is there a valid reason for that?